src/fparser/two/utils.py

```diff
@@ -1308,69 +1308,87 @@ class WORDClsBase(Base):
 
     '''
     @staticmethod
-    def match(pattern, cls, string, check_colons=False, require_cls=False):
-        ''':param pattern: the pattern of the WORD to match. This can be a \
-        string, a list of strings or a tuple of strings.
-        :type pattern: str, tuple of str or list of str.
+    def match(pattern, cls, string, colons=False, require_cls=False):
+        '''Checks whether the content in string matches the expected
+        WORDClsBase format with 'pattern' providing the keyword, 'cls'
+        providing the following text, 'colons' specifying whether an
+        optional '::' is allowed as a separator between the keyword
+        and cls and 'require_cls' specifying whether cls must have
+        content or not.
+
+        Note, if the optional '::' is allowed and exists in the string
+        then 1) cls must also have content i.e. it implies
+        `require_cls=True` and 2) white space is not required between
+        the keyword and the '::' and the '::' and cls.
+
+        The simplest form of pattern is a string. However this method
+        can also match more complex patterns as specified by the
+        Pattern class in pattern_tools.py. As patterns can be built
+        from combinations of other patterns (again see
+        pattern_tool.py) this method also supports a hierarchy of
+        lists and/or tuples of patterns.
+
+        :param pattern: the pattern of the WORD to match. This can be \
+            a Pattern, string, list or tuple, with a list or tuple \
+            containing one or more Pattern, string, list or tuple.
+        :type pattern: :py:class:`fparser.two.pattern_tools.Pattern`, \
+            str, tuple of str/Pattern/tuple/list or list of \
+            str/Pattern/tuple/list
         :param cls: the class to match.
-        :type cls: a subclass of :py:class:`fparser.two.utils.Base`.
+        :type cls: a subclass of :py:class:`fparser.two.utils.Base`
         :param str string: Text that we are trying to match.
-        :param bool check_colons: whether '::' is allowed or not \
-        between WORD and cls.
+        :param bool colons: whether '::' is allowed as an optional \
+            separator between between WORD and cls.
         :param bool require_cls: whether content for cls is required \
-        or not.
+            or not.
+
+        :returns: None if there is no match or, if there is a match, a \
+            2-tuple containing a string matching the 'WORD' and an \
+            instance of 'cls' (or None if an instance of cls is not \
+            required and not provided).
+        :rtype: (str, cls or NoneType) or NoneType
 
         '''
         if isinstance(pattern, (tuple, list)):
             for p in pattern:
```

Comment:
Not strictly your fault but since you've modified the code within this loop, please could you rename `p` to something that pylint is happy about.

Reply:
Done.